cmd/ping.go

```diff
@@ -18,15 +20,42 @@ can connect to Kong's Admin API.`,
 		Args: validateNoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
-
-			wsConfig := rootConfig.ForWorkspace(pingWorkspace)
-			version, err := fetchKongVersion(ctx, wsConfig)
-			if err != nil {
-				return fmt.Errorf("reading Kong version: %w", err)
+			cmdStr := "ping"
+			var version string
+			var err error
+			mode := getMode(nil)
+			if mode == modeKonnect {
+				// get Konnect client
+				httpClient := utils.HTTPClient()
+				konnectClient, err := utils.GetKonnectClient(httpClient, konnectConfig)
+				if err != nil {
+					return err
+				}
+				u, _ := url.Parse(konnectConfig.Address)
+				// authenticate with konnect
+				res, err := authenticate(ctx, konnectClient, u.Host)
+				if err != nil {
+					return fmt.Errorf("authenticating with Konnect: %w", err)
+				}
+				fullName := res.FullName
+				if res.FullName == "" {
+					fullName = fmt.Sprintf("%s %s", res.FirstName, res.LastName)
+				}
+				fmt.Printf("Successfully Konnected as %s (%s)!\n",
+					fullName, res.Organization)
+				if konnectConfig.Debug {
+					fmt.Printf("Organization ID: %s\n", res.OrganizationID)
+				}
+			} else {
+				wsConfig := rootConfig.ForWorkspace(pingWorkspace)
+				version, err = fetchKongVersion(ctx, wsConfig)
+				if err != nil {
+					return fmt.Errorf("reading Kong version: %w", err)
+				}
+				fmt.Println("Successfully connected to Kong!")
+				fmt.Println("Kong version: ", version)
```

Comment:
Can you break into two functions? pingKonnect and pingKong?
Please avoid long functions.